tests/bluetooth/bsim_bt/bsim_test_mesh/src/test_beacon.c

```diff
@@ -229,6 +235,128 @@ static void test_rx_on_key_refresh(void)
 	PASS();
 }
 
+static void test_tx_beacon_init(void)
+{
+	bt_mesh_test_cfg_set(&tx_cfg, BEACON_OBSERVE_INTERVAL);
+}
+
+static void test_rx_beacon_init(void)
+{
+	bt_mesh_test_cfg_set(&tx_cfg, BEACON_OBSERVE_INTERVAL);
+}
+
+static void test_tx_beacon_secure(void)
+{
+	uint32_t delay_time;
+	uint32_t sent_beacons;
+	uint32_t sent_beacon_time;
+	struct bt_mesh_subnet *sub;
+	
+	/* shift beaconing time line to avoid boundary cases. */
+	delay_time = get_device_nbr() * 100;
+	k_sleep(K_MSEC(delay_time));
+
+	bt_mesh_test_setup();
+
+	sub = bt_mesh_subnet_find(NULL, NULL);
+
+	sent_beacons = 0;
+	sent_beacon_time = 0;
+	while (true) {
```

Comment:
It is better to not do an infinite loop in tests. If functionality is broken then the test CI plan will get stuck forever. Just use one more semaphore or any other suitable Zephyr service.

Reply:
If I don't miss anything this while-loop could be replaced by something like this:
```
k_sleep(K_SECONDS(BEACON_OBSERVE_INTERVAL));
ASSERT_TRUE(sub->beacon_sent > 0);
PASS();
```